customer_satisfaction/model/model_dev.py

```diff
@@ -0,0 +1,202 @@
+                trial = study.best_trial
+                n_estimators = trial.params["n_estimators"]
+                max_depth = trial.params["max_depth"]
+                min_samples_split = trial.params["min_samples_split"]
+                print("Best parameters : ", trial.params)
+                reg = RandomForestRegressor(
+                    n_estimators=n_estimators,
+                    max_depth=max_depth,
+                    min_samples_split=min_samples_split,
+                )
+                reg.fit(self.x_train, self.y_train)
+                return reg
+            else:
+                model = RandomForestRegressor(
+                    n_estimators=152, max_depth=20, min_samples_split=17
+                )
+                model.fit(self.x_train, self.y_train)
+                return model
+        except Exception as e:
+            logging.error("Error in training Random Forest model")
+            logging.error(e)
+            return None
+
+
+    def lightgbm_trainer(self, fine_tuning: bool = True): 
+        """
+        It trains the LightGBM model.
+        
+        Args:
+            fine_tuning: If True, hyperparameter optimization is performed. If False, the default
+            parameters are used, defaults to True (optional)
+        """
+
+        logging.info("Entered for training LightGBM model")
+        try:
+            if fine_tuning:
+                hyper_opt = Hyperparameter_Optimization(
+                    self.x_train, self.y_train, self.x_test, self.y_test
+                )
+                study = optuna.create_study(direction="maximize")
+                study.optimize(hyper_opt.optimize_lightgbm, n_trials=100)
+                trial = study.best_trial
+                n_estimators = trial.params["n_estimators"]
+                max_depth = trial.params["max_depth"]
+                learning_rate = trial.params["learning_rate"]
+                reg = LGBMRegressor(
+                    n_estimators=n_estimators,
+                    learning_rate=learning_rate,
+                    max_depth=max_depth,
+                )
+                reg.fit(self.x_train, self.y_train)
+                return reg
+            else:
+                model = LGBMRegressor(
+                    n_estimators=200, learning_rate=0.01, max_depth=20
+                )
+                model.fit(self.x_train, self.y_train)
+                return model
+        except Exception as e:
+            logging.error("Error in training LightGBM model")
+            logging.error(e)
+            return None
+
+    def xgboost_trainer(self, fine_tuning: bool = True):
+        """
+        It trains the xgboost model.
+        
+        Args:
+            fine_tuning: If True, hyperparameter optimization is performed. If False, the default
+            parameters are used, defaults to True (optional)
+        """
+
+        logging.info("Entered for training XGBoost model")

```

Comment:
Tiny thing, but it might be clearer to say "Started training XGBoost ..." etc.

Reply:
Sure